spec/commands/power_spec.rb

```diff
@@ -58,10 +58,30 @@ def run_power(node_identifier, command, **options)
         node_names.each do |name|
           expect(Metalware::SystemCommand).to receive(:run).with(
             "ipmitool -H #{name}.bmc -I lanplus -U bmcuser -P bmcpassword chassis power on"
-          ).ordered
+          ).ordered.and_return('output123')
+        end
+
+        output = run_power('nodes', 'on', gender: true, sleep: 0.5)
+
+        node_names.each do |name|
+          expect(output).to include("#{name}: output123")
         end
+      end
+
+      it 'does not error when individual ipmi commands error' do
+        allow(
+          Metalware::SystemCommand
+        ).to receive(:run)
+          .with(/ipmitool/)
+          .and_raise(Metalware::SystemCommandError, 'error123')
```

Comment:
Good start for testing the command but could we also ensure the command is ran the correct number of times? Just in case the error is short circuiting the loop.

Also can we only raise the error once for a single node?